client/src/panels/panel_declarations/finances/drr_dp_resources/crso_by_prog.js

```diff
@@ -180,29 +309,68 @@ const get_calculate_func = (is_fte) => {
       Both exp and fte data is returned in either case so the render can ensure consistency in the association of colour to program names
       across the two panels
     */
-    const exp_data = _.map(programSpending.q(subject).data, (row) => ({
+    const exp_data = _.map(queried_exp.data, (row) => ({
       label: row.prgm,
-      data: planning_years.map((col) => row[col]),
+      data: exp_years_with_gap_year.map((col) =>
+        _.isUndefined(row[col]) ? null : row[col]
+      ),
     }));
-    const fte_data = _.map(programFtes.q(subject).data, (row) => ({
+    const fte_data = _.map(queried_fte.data, (row) => ({
       label: row.prgm,
-      data: planning_years.map((col) => row[col]),
+      data: fte_years_with_gap_year.map((col) =>
+        _.isUndefined(row[col]) ? null : row[col]
+      ),
     }));
 
-    const relevant_data = is_fte ? fte_data : exp_data;
-    const first_year_top_2_programs = _.chain(relevant_data)
-      .sortBy(({ data }) => _.first(data))
+    const relevant_data = is_fte ? queried_fte.data : queried_exp.data;
+    const most_recent_top_2_programs = _.chain(relevant_data)
+      .flatMap((program_data) => ({
+        prgm: program_data.prgm,
+        value:
+          program_data[is_fte ? "{{pa_last_year}}" : "{{pa_last_year}}exp"],
+      }))
+      .sortBy("value")
+      .takeRight(2)
+      .reverse()
+      .value();
+    const most_recent_planned_top_2_programs = _.chain(relevant_data)
```

Comment:
Small nitpick, but referring to planed spending as "most recent" was a little confusing to me at first. If this is hardcoded to planning year 1 then just call it that IMO.